app/src/main/java/org/dslul/openboard/inputmethod/latin/utils/InlineAutofillUtils.java

```diff
@@ -0,0 +1,119 @@
+package org.dslul.openboard.inputmethod.latin.utils;
+
+import static android.util.TypedValue.COMPLEX_UNIT_DIP;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.graphics.Color;
+import android.graphics.drawable.Icon;
+import android.os.Build;
+import android.os.Bundle;
+import android.util.Size;
+import android.util.TypedValue;
+import android.view.ViewGroup;
+import android.view.inputmethod.InlineSuggestion;
+import android.view.inputmethod.InlineSuggestionsRequest;
+import android.widget.HorizontalScrollView;
+import android.widget.LinearLayout;
+import android.widget.inline.InlinePresentationSpec;
+
+import androidx.annotation.RequiresApi;
+import androidx.autofill.inline.UiVersions;
+import androidx.autofill.inline.common.ImageViewStyle;
+import androidx.autofill.inline.common.TextViewStyle;
+import androidx.autofill.inline.common.ViewStyle;
+import androidx.autofill.inline.v1.InlineSuggestionUi;
+
+import org.dslul.openboard.inputmethod.latin.R;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RequiresApi(api = Build.VERSION_CODES.R)
+public class InlineAutofillUtils {
+
+    private static int toPixel(int dp, Context context) {
+        return (int) TypedValue.applyDimension(COMPLEX_UNIT_DIP, dp,
+                context.getResources().getDisplayMetrics());
+    }
+
+    private static int getHeight(Context context) {
+        return context.getResources().getDimensionPixelSize(R.dimen.config_suggestions_strip_height);
+    }
+
+    public static InlineSuggestionsRequest createInlineSuggestionRequest(Context context) {
```

Comment:
I remember this is called on every text field. Does it have good performance?
If not I consider adding some sort of cache...

Reply:
I did not notice any performance difference, but I don't know how it will perform on older devices that support this feature and I am not sure how to cache it.

Reply:
Cache is no problem, I can add it later if I have the impression it's necessary.
How long does the function take? You can do a simple measurement by adding a log line as first and last line, and check the timestamps.

Reply:
The function takes about 0.001 or 0.002 seconds.

Reply:
Thanks, then I guess it's not worth adding a cache